llama_index/embeddings/gradient.py

```diff
@@ -0,0 +1,95 @@
+import logging
+from typing import Any, List, Optional
+
+from llama_index.bridge.pydantic import Field, PrivateAttr
+from llama_index.embeddings.base import BaseEmbedding, Embedding
+
+logger = logging.getLogger(__name__)
+
+
+# For bge models that Gradient AI provides, it is suggested to add the instruction for retrieval.
+# Reference: https://huggingface.co/BAAI/bge-large-en-v1.5#model-list
+QUERY_INSTRUCTION_FOR_RETRIEVAL = (
+    "Represent this sentence for searching relevant passages:"
+)
+
+
+class GradientEmbedding(BaseEmbedding):
+    """GradientAI embedding models.
+
+    This class provides an interface to generate embeddings using a model
+    deployed in Gradient AI. At the initialization it requires a model_id
+    of the model deployed in the cluster.
+
+    Note:
+        Requires `gradientai` package to be available in the PYTHONPATH. It can be installed with
+        `pip install gradientai`.
+    """
+
+    _gradient: Any = PrivateAttr()
+    _model: Any = PrivateAttr()
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "GradientEmbedding"
+
+    def __init__(
+        self,
+        gradient_access_token: str,
+        gradient_workspace_id: str,
```

Comment:
these should be optional so you can use your env vars

Reply:
done!